scripts/install_submitter.py

```
@@ -0,0 +1,768 @@
+            key = winreg.OpenKey(  # type: ignore
+                winreg.HKEY_CURRENT_USER,
+                "Environment",
+                0,
+                winreg.KEY_SET_VALUE,  # type: ignore
+            )
+            winreg.SetValueEx(key, name, 0, winreg.REG_SZ, value)  # type: ignore
+            winreg.CloseKey(key)  # type: ignore
+            logger.info("Set %s=%s", name, value)
+            self._notify_environment_change()
+            return True
+        except (OSError, ImportError) as e:
+            logger.error("Failed to set environment variable %s: %s", name, e)
+            return False
+
+    def _notify_environment_change(self) -> None:
+        """
+        Notify Windows system of environment variable changes.
+        The system broadcast ensures all running applications and future processes can
+        access the updated env vars immediately. Without this notification, newly launched
+        VRED Pro will not recognize the VRED-specific env vars that this script added,
+        causing the Deadline Cloud submitter to not appear in VRED's interface.
+        """
+        try:
+            # Constants for Win32 API
+            HWND_BROADCAST = 0xFFFF
+            WM_SETTINGCHANGE = 0x001A
+            SMTO_ABORTIFHUNG = 0x0002
+
+            # Send broadcast message to notify all windows of environment change
+            result = ctypes.windll.user32.SendMessageTimeoutW(
+                HWND_BROADCAST,
+                WM_SETTINGCHANGE,
+                0,
+                "Environment",
+                SMTO_ABORTIFHUNG,
+                5000,  # 5 second timeout
+                None,
+            )
+
+            if result:
+                logger.info("Notified system of environment variable changes")
+            else:
+                logger.warning("Failed to notify system of environment variable changes")
+
+        except Exception as e:
+            logger.debug("Could not notify system of environment changes: %s", e)
```

Do we want to raise an exception here? Similar comments here to places where we catch exceptions but don't re-raise them. There's probably places where swallowing the error is fine, because we don't want to abort the whole intallation over a minor issue, but there could be places where we do want to consider stopping everything rather than completing a broken install

---

Thanks, good point. I've updated `_notify_environment_change()` to re-raise exceptions instead of swallowing them. While environment variable notification failure isn't very critical (because, at this point, the environment variables are already set successfully), I think it's better to re-raise the exception for better visibility into potential issues, instead of just logging and continuing silently.
I have also updated other places for error/exception handling, please see my [other comment](https://github.com/aws-deadline/deadline-cloud-for-vred/pull/85#discussion_r2402365470) for details.